cpp/src/parquet/arrow/schema_internal.cc

```diff
@@ -90,7 +90,7 @@ Result<std::shared_ptr<ArrowType>> MakeArrowTime64(const LogicalType& logical_ty
 Result<std::shared_ptr<ArrowType>> MakeArrowTimestamp(const LogicalType& logical_type) {
   const auto& timestamp = checked_cast<const TimestampLogicalType&>(logical_type);
   const bool utc_normalized =
-      timestamp.is_from_converted_type() ? false : timestamp.is_adjusted_to_utc();
+      timestamp.is_from_converted_type() ? true : timestamp.is_adjusted_to_utc();
```

Comment:
CMIW, the change here is to align with cpp/src/parquet/types.cc where is_adjusted_to_utc of logical type is set true.

Reply:
Here, every caller except here is set to true.

By the way, I didn't find Int96 in the doc, would Int96 should having `is_adjusted_to_utc = true` ? (Rust impl set it false)

Reply:
Why not just
```cpp
  const bool utc_normalized = timestamp.is_adjusted_to_utc();
```

Reply:
I think the following would be equivalent to the updated ternary expression:
```cpp
const bool utc_normalized = timestamp.is_from_converted_type() || timestamp.is_adjusted_to_utc();
```

Reply:
I've changed to:

```
  const bool utc_normalized = timestamp.is_adjusted_to_utc();
```

I think this is enough

Reply:
> Here, every caller except here is set to true. See: https://github.com/apache/arrow/blob/main/cpp/src/parquet/schema.cc#L443
> 
> By the way, I didn't find Int96 in the doc, would Int96 should having `is_adjusted_to_utc = true` ? (Rust impl set it false)

IMO, INT96 is a physical type and orthogonal to logical/converted type.

Reply:
> INT96 is a physical type and orthogonal to logical/converted type.

Yeah, currently when we meet int96, we will bindly casting it to time-related, `!adjustedToUtc` value

Reply:
@joellubi :

https://github.com/apache/arrow/blob/main/cpp/src/parquet/schema.cc#L443 The thrift schema would be casted here, and then goes to `LogicalType::FromConvertedType`. It would set `adjustedToUtc = true`: https://github.com/apache/arrow/blob/main/cpp/src/parquet/types.cc#L347-L358

So I think just `timestamp.is_adjusted_to_utc()` is ok